test/StripTagsTest.php

```diff
@@ -511,6 +511,34 @@ public function testMultiQuoteInput()
         $this->assertEquals($expected, $filter->filter($input));
     }
 
+    public function badCommentProvider()
+    {
+        return [
+            ['A <!--> B', 'A '],  // Should be treated as just an open
+            ['A <!---> B', 'A '], // Should be treated as just an open
+            ['A <!----> B', 'A  B'],
+            ['A <!-- --> B', 'A  B'],
+            ['A <!--> B <!--> C', 'A  C'],
+            ['A <!-- -- > -- > -->', 'A '],
+            ["A <!-- B\n C\n D -->", 'A '],
```

Comment:
In this two above tests we should have something after `-->` so we know that not the whole content after opening is removed.

Reply:
Done.